contrib/utilities/indent-branch

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2018 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# This script does the same thing as contrib/utilities/indent but only reformats
+# files which have changed (or been added and not committed) since the last
+# merge commit to the master branch. It should be run as
+#
+#    make indent-branch
+#
+# from a build directory.
+#
+
+if test ! -d source -o ! -d include -o ! -d examples ; then
+  echo "*** This script must be run from the top-level directory of deal.II."
+  exit 1
+fi
+
+# check that clang-format is available:
+source contrib/utilities/indent-checks
+
+# Filenames may (in theory) may consist of any characters besides '/' and NUL
+# (including newlines, terminal commands, etc). Hence, this script uses NUL as
+# the file name separation character. Since bash cannot store NUL-delimited
+# output in variables we convert it back-and-forth from base64.
+
+# add any new files:
+UNSORTED_FILES=$(git ls-files -z --others --exclude-standard | base64)
+# add any files changed relative to the last merge commit to master:
+LAST_MERGE_COMMIT=$(git log --format="%H" --merges --max-count=1 master)
+UNSORTED_FILES+=$(git diff -z --name-only $LAST_MERGE_COMMIT | base64)
+# uniquify:
+SORTED_FILES=$(base64 -d <<< $UNSORTED_FILES | sort -z | uniq -z | base64)
+
+SOURCE_REGEX='\.cc$\|\.h$\|\.cu$\|\.cuh$'
+
+# Only update the file if the version produced by clang-format is different.
+maybe_reindent_file()
+{
+    # mktemp works differently in macOS and GNU/Linux: in particular macOS needs
+    # the template
+    TEMP_DIRECTORY=$(mktemp -d 2>/dev/null || mktemp -d -t "deal")
+    indented_file=$(mktemp $TEMP_DIRECTORY/XXXX.tmp.cc)
+
+    clang-format $1>$indented_file
+    if ! diff -q $indented_file $1 >/dev/null
+    then
+        mv $indented_file $1
+    fi
+    rm -r $TEMP_DIRECTORY
+}
+export -f maybe_reindent_file
+
+# 1. Reindent:
+base64 -d <<< $SORTED_FILES | grep -z $SOURCE_REGEX  \
+    | xargs --no-run-if-empty --max-procs=0 -0 -I {} \
+            bash -c 'maybe_reindent_file "$@"' _ {}
+
+# 2. Fix permissions:
+base64 -d <<< $SORTED_FILES | grep -z $SOURCE_REGEX \
+    | xargs --no-run-if-empty --max-procs=0 -0 chmod 644
+
+# 3. Convert DOS to UNIX:
+dos_to_unix()
```

Comment:
I think this and maybe a number of other things in this script could also go to the common file.